tests/HUBComponentModelBuilderTests.m

```diff
@@ -592,4 +592,88 @@ - (void)testCopying
     XCTAssertEqualObjects(copiedCustomImageDataBuilder.placeholderIconIdentifier, @"customPlaceholder");
 }
 
+- (void)testBuildersForChildrenInGroupWhenAddingChildBuilder
+{
+    NSString * const firstChildModelIdentifier = @"firstIdentifier";
+    NSString * const secondChildModelIdentifier = @"secondIdentifier";
+    NSString * const firstChildGroupIdentifier = @"firstGroup";
+    NSString * const secondChildGroupIdentifier = @"secondGroup";
+
+    id<HUBComponentModelBuilder> const childBuilder = [self.builder builderForChildWithIdentifier:firstChildModelIdentifier];
+    childBuilder.groupIdentifier = firstChildGroupIdentifier;
+    id<HUBComponentModelBuilder> const childBuilder2 = [self.builder builderForChildWithIdentifier:secondChildModelIdentifier];
+    childBuilder2.groupIdentifier = secondChildGroupIdentifier;
+
+    NSArray *buildersForChildrenInGroup = [self.builder buildersForChildrenInGroupWithIdentifier:firstChildGroupIdentifier];
+    XCTAssertTrue(buildersForChildrenInGroup.count == 1);
+    XCTAssertEqualObjects(buildersForChildrenInGroup.firstObject, childBuilder);
+}
+
+- (void)testBuildersForChildrenInGroupWhenRemovingChildBuilder
+{
+    NSString * const firstChildModelIdentifier = @"firstIdentifier";
+    NSString * const secondChildModelIdentifier = @"secondIdentifier";
+    NSString * const firstChildGroupIdentifier = @"firstGroup";
+    NSString * const secondChildGroupIdentifier = @"secondGroup";
+
+    id<HUBComponentModelBuilder> const childBuilder = [self.builder builderForChildWithIdentifier:firstChildModelIdentifier];
+    childBuilder.groupIdentifier = firstChildGroupIdentifier;
+    id<HUBComponentModelBuilder> const childBuilder2 = [self.builder builderForChildWithIdentifier:secondChildModelIdentifier];
+    childBuilder2.groupIdentifier = secondChildGroupIdentifier;
+
+    [self.builder removeBuilderForChildWithIdentifier:firstChildModelIdentifier];
+
+    NSArray *buildersForFirstGroup = [self.builder buildersForChildrenInGroupWithIdentifier:firstChildGroupIdentifier];
+    NSArray *buildersForSecondGroup = [self.builder buildersForChildrenInGroupWithIdentifier:secondChildGroupIdentifier];
+
+    XCTAssertTrue(buildersForFirstGroup.count == 0);
```

Comment:
How about using `XCTAssertNil` instead, since that's what the API tells us that it will return.